### DISCUSSION TITLE
Proposal: Using Harberger Tax in Global State Machine

### DISCUSSION THREAD

**Discussion by geopor** [tearust/teaproject#126] [Ideas]
This is a discussion for the new Harberger Tax proposal that will be applied to global CML nodes.

The post is here: https://teaproject.medium.com/proposal-using-a-harberger-tax-in-the-global-state-machine-4323976f9b8

**Comment by geopor** [comment]
One issue I see with the **state memory tax** is when demand goes higher, then the price of these state memory tokens as well as the taxes paid everyday will increase. That will push a hyper-competitive TApp marketplace where only the most profitable TApps will survive. Although that sounds like textbook capitalism, we have to remember that all apps have to start somewhere. If it's too expensive for small, not-yet-profitable apps to run on our platform, then we risk stifling innovation and an exodus of young developers to competing decentralized platforms.

**Comment by kevingzhang** [comment]
younger smaller applications do not necessarily mean less profitable than older larger applications. Consider the cost of applications are based on resource usage. If a larger application has lower ARPU, the profit is lower than a smaller application that has higher ARPU.

**Reply by kevingzhang** [reply]
ARPU is the average end-user revenue.
For example, If TEA Party is a paid application that focuses on high-value customers, Twitter is a general-purpose application. If they are using the same business model, (I know they are not, but assume they are), Twitter has much higher cost than TEA Party. So TEA Party is more likely to survive than Twitter

**Comment by kevingzhang** [comment]
@geopor I suggest you post this discussion to twitter so that more community members can join the discussion

**Reply by geopor** [reply]
Ok posted https://twitter.com/teaprojectorg/status/1538960126148632577
